src/Microsoft.Dnx.Host.Clr/DomainManager.cs

```diff
@@ -36,12 +36,34 @@ public override void InitializeNewDomain(AppDomainSetup appDomainInfo)
             Environment.SetEnvironmentVariable(EnvironmentNames.AppBase, _info.ApplicationBase);
         }
 
-        var version = DetermineDnxVersion(_info.ApplicationBase);
+        appDomainInfo.ApplicationBase = Environment.GetEnvironmentVariable(EnvironmentNames.DefaultLib);
+        appDomainInfo.TargetFrameworkName = DetermineAppDomainTargetFramework();
+    }
+
+    private string DetermineAppDomainTargetFramework()
+    {
+        // Check if the native bootstrapper gave us a framework name
+        string frameworkName = Environment.GetEnvironmentVariable(EnvironmentNames.Framework);
+        FrameworkName framework;
+        Version version;
+        if (!string.IsNullOrEmpty(frameworkName) &&
+            FrameworkNameUtility.TryParseFrameworkName(frameworkName, out framework) &&
+            framework.Identifier.Equals(FrameworkNames.LongNames.Dnx, StringComparison.OrdinalIgnoreCase))
+        {
+            // It's a DNX framework! So just use that version as the .NET version
+            version = framework.Version;
+        }
+        else
+        {
```

Comment:
Do we want to log something when `frameworkName` is not empty but we fail to parse it?